drivers/watchdog/wdt_nrfx.c

```diff
@@ -28,40 +28,49 @@ static int wdt_nrf_setup(const struct device *dev, uint8_t options)
 {
 	const struct wdt_nrfx_config *config = dev->config;
 	struct wdt_nrfx_data *data = dev->data;
-	uint32_t behaviour;
 
 	/* Activate all available options. Run in all cases. */
-	behaviour = NRF_WDT_BEHAVIOUR_RUN_SLEEP_MASK | NRF_WDT_BEHAVIOUR_RUN_HALT_MASK;
+	config->config.behaviour = NRF_WDT_BEHAVIOUR_RUN_SLEEP_MASK |
+#if NRF_WDT_HAS_STOP
+				   NRF_WDT_BEHAVIOUR_STOP_ENABLE_MASK |
+#endif
+				   NRF_WDT_BEHAVIOUR_RUN_HALT_MASK;
 
 	/* Deactivate running in sleep mode. */
 	if (options & WDT_OPT_PAUSE_IN_SLEEP) {
-		behaviour &= ~NRF_WDT_BEHAVIOUR_RUN_SLEEP_MASK;
+		config->config.behaviour &= ~NRF_WDT_BEHAVIOUR_RUN_SLEEP_MASK;
 	}
 
 	/* Deactivate running when debugger is attached. */
 	if (options & WDT_OPT_PAUSE_HALTED_BY_DBG) {
-		behaviour &= ~NRF_WDT_BEHAVIOUR_RUN_HALT_MASK;
+		config->config.behaviour &= ~NRF_WDT_BEHAVIOUR_RUN_HALT_MASK;
 	}
 
-	nrf_wdt_behaviour_set(config->wdt.p_reg, behaviour);
-	/* The watchdog timer is driven by the LFCLK clock running at 32768 Hz.
-	 * The timeout value given in milliseconds needs to be converted here
-	 * to watchdog ticks.*/
-	nrf_wdt_reload_value_set(
-		config->wdt.p_reg,
-		(uint32_t)(((uint64_t)data->m_timeout * 32768U)
-			   / 1000));
+	config->config.reload_value = data->m_timeout;
 
+	nrfx_wdt_reconfigure(&config->wdt, &config->config);
 	nrfx_wdt_enable(&config->wdt);
 
 	return 0;
 }
 
 static int wdt_nrf_disable(const struct device *dev)
 {
-	/* Started watchdog cannot be stopped on nRF devices. */
+#if NRFX_WDT_HAS_STOP
+	const struct wdt_nrfx_config *config = dev->config;
+	nrfx_err_t err_code;
+
+	err_code = nrfx_wdt_stop(&config->wdt);
+
+	if (err_code == NRFX_SUCCESS) {
+		return 0;
+	}
+
+	return -ENOTSUP;
```

Comment:
nit: other functions in this driver return errors from `if`s and success at the end; it would be nice to be consistent in this regard.